src/otx/engine/engine.py

```diff
@@ -904,3 +915,19 @@ def _get_anomaly_model(
         model.optimizer = optimizer
         model.scheduler = scheduler
         return model
+
+    def _patch_packages_xpu(self) -> None:
+        """Patch packages when xpu is available."""
+        if self.task in [OTXTaskType.DETECTION, OTXTaskType.INSTANCE_SEGMENTATION] and self.device.accelerator == "xpu":
+            import numpy as np
+            from mmcv.ops.nms import NMSop
+            from mmcv.ops.roi_align import RoIAlign
+            from mmengine.structures import instance_data
+
+            from otx.algo.detection.utils import monkey_patched_nms, monkey_patched_roi_align
+
+            long_type_tensor = Union[torch.LongTensor, torch.xpu.LongTensor]
+            bool_type_tensor = Union[torch.BoolTensor, torch.xpu.BoolTensor]
+            instance_data.IndexType = Union[str, slice, int, list, long_type_tensor, bool_type_tensor, np.ndarray]
```

Comment:
It's just a question. Why is it needed?

Reply:
Unfortunately, if we won't patch this -> mmengine will raise an error during bbox post processing since XPU device is not supported by MMLab. It didn't happen in OTX1.6 since we used old version of mm* libraries.